File: 94 Binary Tree Inorder Traversal/94.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    let res = []
    let stack = []
    let cur = root
    while (cur || stack.length) {
        if (cur) {
            stack.push(cur)
            cur = cur.left
        } else {
            let temp = stack.pop()
            res.push(temp.val)
            cur = temp.right
        }
    }
    return res;
};

File: 94 Binary Tree Inorder Traversal/94.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
from collections import deque
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        res = []
        stack = deque()
        cur = root
        while cur or stack:
            if cur:
                stack.append(cur)
                cur = cur.left
            else:
                temp = stack.pop()
                res.append(temp.val)
                cur = temp.right
        return res